File: TI1.1SOAP_JAVA_SINBDD_GR04/02.CLICON/ConUni_SOAP_Java_G04_Cliente_Consola/src/ec/edu/monster/servicios/ConversorService.java

package ec.edu.monster.servicios;



/**
 *
 * @author Rabedon1
 */
public class ConversorService {

    public static double convertirLongitud(double valor, java.lang.String unidadOrigen, java.lang.String unidadDestino) {
        ec.edu.monster.ws.ConversorService_Service service = new ec.edu.monster.ws.ConversorService_Service();
        ec.edu.monster.ws.ConversorService port = service.getConversorServicePort();
        return port.convertirLongitud(valor, unidadOrigen, unidadDestino);
    }

    public static double convertirMasa(double valor, java.lang.String unidadOrigen, java.lang.String unidadDestino) {
        ec.edu.monster.ws.ConversorService_Service service = new ec.edu.monster.ws.ConversorService_Service();
        ec.edu.monster.ws.ConversorService port = service.getConversorServicePort();
        return port.convertirMasa(valor, unidadOrigen, unidadDestino);
    }

    public static double convertirTemperatura(double valor, java.lang.String unidadOrigen, java.lang.String unidadDestino) {
        ec.edu.monster.ws.ConversorService_Service service = new ec.edu.monster.ws.ConversorService_Service();
        ec.edu.monster.ws.ConversorService port = service.getConversorServicePort();
        return port.convertirTemperatura(valor, unidadOrigen, unidadDestino);
    }

   
      
}

File: TI1.3RESTFUL_JAVA_SINBDD_GR04/02.CLIMOV/ConUni_Soap_Java_Cliente_Movil_G04/app/src/main/java/com/example/conuni_soap_java_g04_cliente_movil/ec/edu/monster/servicios/ConversorService.kt
package com.example.conuni_soap_java_g04_cliente_movil.ec.edu.monster.servicios


import com.example.conuni_soap_java_g04_cliente_movil.ec.edu.monster.modelo.Unidad
import com.example.conuni_soap_java_g04_cliente_movil.ec.edu.monster.servicios.soap.ConversorSoapClient
import java.util.Locale

class ConversorService(
    private val soap: ConversorSoapClient = ConversorSoapClient()
) {
    fun longitudes(valor: Double, de: Unidad, a: Unidad) =
        format(valor, de, soap.convertirLongitud(valor, de, a), a)

    fun masas(valor: Double, de: Unidad, a: Unidad) =
        format(valor, de, soap.convertirMasa(valor, de, a), a)

    fun temperaturas(valor: Double, de: Unidad, a: Unidad) =
        format(valor, de, soap.convertirTemperatura(valor, de, a), a)

    private fun format(v: Double, de: Unidad, r: Double, a: Unidad): String {
        fun lbl(u: Unidad) = when (u) {
            Unidad.METROS -> "Metros"; Unidad.KILOMETROS -> "Kilómetros"
            Unidad.MILLAS -> "Millas"; Unidad.YARDAS -> "Yardas"
            Unidad.GRAMOS -> "Gramos"; Unidad.KILOGRAMOS -> "Kilogramos"
            Unidad.LIBRAS -> "Libras"; Unidad.ONZAS -> "Onzas"
            Unidad.CELSIUS -> "°C"; Unidad.FAHRENHEIT -> "°F"; Unidad.KELVIN -> "K"
        }
        fun num(x: Double) = String.format(Locale.US, "%.4f", x)
        return "${num(v)} ${lbl(de)} = ${num(r)} ${lbl(a)}"
    }
}